### PR TITLE
Bump JSZip to the latest version

### PR DESCRIPTION
**Pull request by takahirox**
Let's update `examples/js/libs/jszip.min.js` to the latest version.

Its API has changed, mainly from sync methods to async ones.
So I also needed to update `3MF/AMF/KMZLoader` and `Editor` where they use `JSZip`.

https://stuk.github.io/jszip/documentation/upgrade_guide.html

Advantage
- With async methods, zip (un)compression doesn't block

Disadvantage
- `jszip.min.js` size increases a bit, from 73KB to 100KB
- new `JSZip` requires `Promises`. (Update) With `JSZip.external.Promise`, non-Promise-support browser doesn't need to load extra `Promise` polyfill
- those loaders `parse` methods will be async from sync

I'm trying `GLTFLoader/Exporter` zip archive support. I wanna work with the latest `JSZip`.

### CODE REVIEW

**Comment by moraxy** [comment]
The upgrade guide says:
> async and loadAsync use (a polyfill of) promises

but you're also referencing the global `Promise` object occasionally, does that work in IE11 and below? (as they don't support `Promise`/`async` yet)

**Comment by moraxy** [comment]
found myself an IE to test with and it breaks [here](https://github.com/takahirox/three.js/blob/3710eb4908eb679a192aab19fc2b023ec6ab3348/examples/js/loaders/3MFLoader.js#L50-L54):
```js
if ( window.TextDecoder === undefined ) {

	return Promise.reject( new Error( 'THREE.ThreeMFLoader: TextDecoder not present. Please use a TextDecoder polyfill.' ) );

}
```
complaining that `Promise` is undefined. If I remove that `if` clause, no more errors come up - but nothing else happens either.

File: examples/js/loaders/KMZLoader.js [hunk 1]
```diff
@@ -20,71 +20,103 @@ THREE.KMZLoader.prototype = {
...
+			scope.parse( text ).then( onLoad ).catch( onError );
 
 		}, onProgress, onError );
 
 	},
 
 	parse: function ( data ) {
 
-		var zip = new JSZip( data ); // eslint-disable-line no-undef
+		if ( JSZip === undefined ) {
```
Comment (donmccurdy) [inline examples/js/loaders/KMZLoader.js]: This will throw a ReferenceError, I think you want `typeof JSZip === 'undefined'`.

Comment (takahirox) [reply examples/js/loaders/KMZLoader.js]: Oops, I've misread the console error message. Thanks!

**Comment by takahirox** [comment]
@moraxy what `Promise` polyfill do you use?

**Comment by moraxy** [comment]
@takahirox oh none so far, I've just run it through IE11 and that's it. But `JSZip` exposes its `Promise` polyfill under `JSZip.external.Promise`, using that should be enough, I hope.

**Comment by takahirox** [comment]
Ah, OK. I understand what you meant now.

Hm, seems like `JSZip.external.Promise === Promise` is `true` on the Promise support browser so prob we should use `JSZip.external.Promise` instead of global `Promise`?

**Comment by moraxy** [comment]
that should be fine, I think

**Comment by takahirox** [comment]
Updated.

**Comment by takahirox** [comment]
Fixed conflicts.

**Comment by Mugen87** [comment]
It seems @mrdoob wants to retain the loader's synchronous parsing pattern:
```js
const loader = new THREE.3MFLoader();
const model = loader.parse( data );
```
So as mentioned here https://github.com/mrdoob/three.js/issues/11583#issuecomment-316280304, we stick to the project's current version or JSZip. Or switch to a library that allows synch reading.

**Pull request closed without merging**